### DISCUSSION TITLE
Got __anchor_private_gen_idl_type when implement wormhole

### DISCUSSION THREAD

**Discussion by linuxhjkaru** [otter-sec/anchor#2988] [Q&A]
From Wormhole Scaffolding, when I build project with anchor 0.29.0.
Then below error shows

```
1481 | pub struct WrappedMint(anchor_spl::token::Mint);
     | -------------------------- associated item `__anchor_private_gen_idl_type` not found for this struct
     |
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following trait defines an item `__anchor_private_gen_idl_type`, perhaps you need to implement it:
             candidate #1: `anchor_lang::IdlBuild`
```

Based on this [PR](https://github.com/coral-xyz/anchor/pull/2011/) , seems like I have to implement __anchor_private_gen_idl_type by myself but how can I do it?

**Comment by acheroncrypto** [comment]
You can add an empty `IdlBuild` implementation for your type which will make it compile, but the type won't get included in the IDL.

```rs
#[cfg(feature = "idl-build")]
impl anchor_lang::IdlBuild for WrappedMint {}
```

See `IdlBuild`'s [documentation](https://docs.rs/anchor-lang/0.30.0/anchor_lang/trait.IdlBuild.html).
